embeddings/config/flair_config.py

```diff
@@ -0,0 +1,174 @@
+            "FlairDocumentPoolEmbedding": {"pooling", "fine_tune_mode"},
+        }
+    )
+
+    @classmethod
+    def map_load_model_keys(cls, document_embedding_cls: str) -> Set[str]:
+        return cls.LOAD_MODEL_KEYS_MAPPING[document_embedding_cls]
+
+    @classmethod
+    def get_load_model_keys(cls) -> Set[str]:
+        return set(chain(*cls.LOAD_MODEL_KEYS_MAPPING.values()))
+
+
+@dataclass
+class FlairBasicConfig(BasicConfig):
+    learning_rate: float = 1e-3
+    mini_batch_size: int = 32
+    max_epochs: int = 20
+
+    task_model_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+    task_train_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+
+    def __post_init__(self) -> None:
+        self.task_train_kwargs = self._parse_fields(self.get_task_train_keys())
+
+    @staticmethod
+    def get_task_train_keys() -> Set[str]:
+        task_train_kwargs = {
+            key for key in FlairBasicConfig.__annotations__.keys() if not key.endswith("_kwargs")
+        }
+        return task_train_kwargs
+
+    @classmethod
+    def from_yaml(cls, path: T_path) -> "FlairBasicConfig":
+        config = read_yaml(path)
+        return cls(**config)
+
+    @classmethod
+    def from_dict(cls, config: Dict[str, Any]) -> "FlairBasicConfig":
+        return cls(**config)
+
+
+@dataclass
+class FlairSequenceLabelingBasicConfig(FlairBasicConfig):
+    hidden_size: int = 256
+    use_crf: bool = True
+    use_rnn: bool = True
+    rnn_type: str = "LSTM"
+    rnn_layers: int = 1
+    dropout: float = 0.0
+    word_dropout: float = 0.05
+    locked_dropout: float = 0.5
+    reproject_embeddings: bool = True
+
+    def __post_init__(self) -> None:
+        self.task_model_kwargs = self._parse_fields(self.get_task_model_keys())
+        super().__post_init__()
+
+    @classmethod
+    def get_task_model_keys(cls) -> Set[str]:
+        task_model_keys = set(FlairSequenceLabelingBasicConfig.__annotations__.keys())
+        task_model_keys.remove("hidden_size")
+        return task_model_keys
+
+
+@dataclass
+class FlairTextClassificationBasicConfig(FlairBasicConfig, FlairTextClassificationConfigMapping):
+    document_embedding_cls: str = "FlairDocumentPoolEmbedding"
+    pooling: str = "mean"
+    fine_tune_mode: str = "none"
+    fine_tune: bool = False
+    cnn_pool_kernels: Tuple[Tuple[int, int], ...] = ((100, 3), (100, 4), (100, 5))
+    hidden_size: int = 256
+    rnn_type: str = "LSTM"
+    rnn_layers: int = 1
+    bidirectional: bool = True
+    dropout: float = 0.0
+    word_dropout: float = 0.05
+    reproject_words: bool = True
+
+    load_model_kwargs: Dict[str, Any] = field(init=True, compare=False, default_factory=dict)
+
+    def __post_init__(self) -> None:
+        load_model_keys = self.get_map_load_model_keys(self.document_embedding_cls)
+        self.load_model_kwargs = self._parse_fields(load_model_keys)
+        super().__post_init__()
```

Comment:
I changed the configs accordingly to your comment.  Not the readability was the problem (IMO it is fine to use post_init to initialize fields based on other attributes as say documentation) but the fact that the config can be dynamically changed by the user and property will reflect the change; otherwise we would have to create new config or manually change the kwargs fields